uk.co.compucorp.civicrm.hrleaveandabsences/js/test/leave-type-wizard/components/leave-type-wizard.component.spec.js

```diff
@@ -213,6 +238,96 @@ define([
             expect(controller.isOnSectionLastTab).toEqual(true);
           });
         });
+
+        describe('when user cancels the form filling', function () {
+          beforeEach(function () {
+            controller.openSection(0);
+            controller.previousTabHandler();
+            $rootScope.$digest();
+          });
+
+          it('redirects to the leave types list page', function () {
+            expect($window.location.href).toBe(
+              sampleURLOrigin + leaveTypeListPageURL);
+          });
+        });
+
+        describe('when user submits the whole wizard form', function () {
+          describe('basic tests', function () {
+            beforeEach(function () {
+              absenceTypeSaverSpy.and.returnValue($q.resolve());
+              fillWizardIn();
+              submitWizard();
+            });
+
+            it('is loading', function () {
+              expect(controller.loading).toBe(true);
+            });
+
+            describe('when saving finishes', function () {
+              beforeEach(function () {
+                $rootScope.$digest();
+              });
+
+              it('saves absence type', function () {
+                var params = AbsenceType.save.calls.mostRecent().args[0];
+
+                expect(AbsenceType.save).toHaveBeenCalled();
+                expect(params).toEqual(jasmine.objectContaining({
+                  title: jasmine.any(String)
+                }));
+                expect(params.carry_forward_expiration_duration_switch).toBeUndefined();
+              });
+
+              it('redirects to the leave types list page', function () {
+                expect($window.location.href).toBe(
+                  sampleURLOrigin + leaveTypeListPageURL);
```

Comment:
make sure to mock the `CRM.url` response here. Or use it directly if it works.

Reply:
Done.